### DISCUSSION TITLE
Adding new pages with different search indexers

### DISCUSSION THREAD

**Discussion by nubgamerz** [microsoft/sample-app-aoai-chatGPT#93] [Ideas]
Hi

I'm customising this as a deployment that we're looking to use against our documentations.

In our search services we have multiple indexers against different sources.

It wouldn't be difficult to add more variables and update the flask app.py, but I'm not familiar with react (I assume it's react?), or tsx.

Is there any way to add new pages that are accessible which would load new chat pages referencing different search indexers?

I would also like to add new pages to add things about Responsible AI, disclaimers and the likes as well.

Is this possible? Thanks

**Comment by pamelafox** [comment]
Hi @nubgamerz - The frontend for this sample is currently written in React, yes. If you prefer using Jinja-based templates, you could likely adapt the frontend I wrote for the ChatGPT quickstart sample: https://github.com/Azure-Samples/chatgpt-quickstart/blob/main/src/flaskapp/templates/index.html
I wrote a bit about that sample here:
http://blog.pamelafox.org/2023/05/streaming-chatgpt-with-server-sent.html
You'd need to add citations to the frontend, and probably other changes.

**Comment by rabbyn** [comment]
Hi @pamelafox , I just forked this repo and made minor changes on the frontend (*) then I changed the connection in the Deployment Center from my Azure WebApp and resync from my forked repo .....after the resync is completed the application is not displaying anymore anything anymore. I'm not a frontend deve and ever did such things before so I'm wondering if I missed something  :p  Thanks

![image](https://github.com/microsoft/sample-app-aoai-chatGPT/assets/72968975/f47aba2a-7ff1-4662-984e-77494e68b1c5)


(*) - Modified Chat.tsx to update the chat title and subtitle
- Modified Layout.tsx to update the header title

**Comment by pamelafox** [comment]
@Mr-Guillaume I haven't used External Git as a deployment mechanism before, so I can't comment as to how that deployment works. I typically use azd, which is detailed here: https://github.com/microsoft/sample-app-aoai-chatGPT/blob/main/README_azd.md

Regardless of the deployment mechanism, the strategy for debugging deploys should be similar. Can you follow the tips on my blog post here? https://blog.pamelafox.org/2023/01/tips-for-debugging-django-app.html
It shows you where to look for debug logs. That blog post is about Django apps but Flask app debugging is similar. 
Let me know if you find any logs and need help interpreting them.

**Comment by sarah-widder** [comment]
As this is not a direct issue with this repo, I'm converting this thread to a discussion. Feel free to continue sharing ideas there!
